### PR TITLE
[14.0][ADD] stock_restrict_expiration_date

### PR DESCRIPTION
**Pull request by petrus-v**
This restrict reservation on expiration date provide while running procurement

This is similar to stock_restrict_lot but instead restrict by expiration date. (We have multiple lot with the same DLC, it's fine to us get any of them)

This is a base module for others.

### CODE REVIEW

**Comment by petrus-v** [comment]
Not clear to me what's worng with codecov/projet :thinking:

File: stock_restrict_expiration_date/models/stock_quant.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        quants = super()._gather(*args, **kwargs)
+        restrict_expiration_date = self.env.context.get("restrict_expiration_date")
+        if restrict_expiration_date:
+            quants = quants.filtered(
+                lambda quant, restrict_expiration_date=restrict_expiration_date: quant.lot_id
+                is False
+                or quant.lot_id
+                and quant.lot_id.expiration_date.date() == restrict_expiration_date
+            )
+        return quants
```
Comment (alexandregaldeano) [inline stock_restrict_expiration_date/models/stock_quant.py:21]: **suggestion,nitpick:**

```suggestion
        if not restrict_expiration_date:
            return quants
        return quants.filtered(
            lambda quant, restrict_expiration_date=restrict_expiration_date: quant.lot_id
            is False
            or quant.lot_id
            and quant.lot_id.expiration_date.date() == restrict_expiration_date
        )
```

**Review by alexandregaldeano — APPROVED** [review]
LGTM

**Review by damdam-s — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**